### ISSUE TITLE
Title/Authors are overwritten using the PDF metadata version

### ISSUE THREAD

**Issue by lfoppiano** [kermitt2/grobid#484]
I'm investigating some issues with duplicated text when disabling the heuristic approach for header extraction and I've noticed the following, in `FullTextParser.java:148`:  

``` 
                    if (metadata.getTitle() != null)
                        resHeader.setTitle(metadata.getTitle());
```

In principle it make sense, but in this case I get `Absence of pseudogap in heavily overdoped Bi 2 Sr 2 CaCu 2 O 8+δ from tunneling spectroscopy of break junctions` overwritten by `7074.dvi`

Is not a trivial task how to merge the two information, I wonder whether there is a better approach. The same happens to authors, btw.

Original PDF: 
[EPL0580589-CC.pdf](https://github.com/kermitt2/grobid/files/3514576/EPL0580589-CC.pdf)

**Comment by lfoppiano** [comment]
Maybe the idea was to override only if `resHeader.getTitle()` was null or empty?

**Comment by kermitt2** [comment]
Yes that's the idea, title and authors are set via the provided pdf embedded metadata in case the other approaches failed. In average with PMC 1942 dataset, it helps a bit, it's why it has been introduced. 

Maybe with better reading order and a reviewed header model, this won't be relevant anymore in the future. 

About the criteria to apply this fall back, we have:

```
if ((resHeader.getTitle() == null) || (resHeader.getTitle().trim().length() == 0) ||
                 (resHeader.getAuthors() == null) || (resHeader.getFullAuthors() == null) ||
                 (resHeader.getFullAuthors().size() == 0) ) 
```

So when there is no author extracted, we use this as a sign of likely failure of the title too, and the embedded metadata are used for the title. This might not be a safe approach in general, through it helps with the evaluation PMC subset (PDF metadata in PMC might be better quality than in average).

**Comment by lfoppiano** [comment]
Yes, the problem is that just after that line, `ResHeader` is updated: 

```
resHeader = new BiblioItem();
                parsers.getHeaderParser().processingHeaderSection(config.getConsolidateHeader(), doc, resHeader);
```

and it could contains the correctly extracted title. 

My idea was to check again if `resHeader.getTitle()` is null before overwriting with the metadata, something like: 

```
if (metadata.getTitle() != null && resHeader.getTitle() == null)
                        resHeader.setTitle(metadata.getTitle());
```

I don't know whether this make sense, there might be also better solutions. I guess this si usually never happening because the heuristic is setting quite often a title and authors.

**Comment by de-code** [comment]
I was going to raise a new issue but then again it's maybe the same issue as this...

Summary of my issue is, that we always replace it with the metadata, even if the metadata is blank.

Example document: 
[212274.full.nometa.pdf](https://github.com/kermitt2/grobid/files/3556265/212274.full.nometa.pdf)
(Source: https://www.biorxiv.org/content/10.1101/212274v1 - PDF metadata removed)

Most PDF documents seem to have some sort of metadata object (especially when converted from docx). But the fields may be blank.

Before, the heuristics would have usually provided the title and it wouldn't have stepped into [the following block](https://github.com/kermitt2/grobid/blob/25258577dcc3b409b20341f19eea56360191ad3d/grobid-core/src/main/java/org/grobid/core/engines/FullTextParser.java#L140-L142):
```java
            if ((resHeader.getTitle() == null) || (resHeader.getTitle().trim().length() == 0) ||
                 (resHeader.getAuthors() == null) || (resHeader.getFullAuthors() == null) ||
                 (resHeader.getFullAuthors().size() == 0) ) {
```

With disabled heuristics we always do. But even if the heuristics failed and we stepped into the fallback. It would mean, if there is a metadata object, we would replace the title with the metadata title even if it was blank. Ignoring the title extracted by the header section.

One simple fix would be to either:
1. use `StringUtils.isEmpty` as a fix; or `isBlank` (if we trust the metadata more)
2. or not override it, if we extracted a title (my preference, maybe could be a configuration)

I believe option 2) this is basically what Luca suggested.

I would suggest, that the metadata check was also moved out of the block.
